TimedArithmeticQuiz/src/components/QuestionHeader.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect, useState } from "react";
+import { quizConfig } from "../data/quizConfig";
+import PropTypes from "prop-types";
+
+export const QuestionHeader = React.memo(function QuestionHeader({ questionNumber, onCountDownEnd }) {
+  const [secondsLeft, setSecondsLeft] = useState(quizConfig.perQuestionTimeInSecs);
+  const [countDownCompleted, setCountDownCompleted] = useState(false);
+
+  useEffect(() => {
+    if (countDownCompleted) {
+      setSecondsLeft(quizConfig.perQuestionTimeInSecs);
+      onCountDownEnd();
+      setCountDownCompleted(false);
```

Comment:
What is the purpose of this LOC.

Reply:
To reset the countdown completed to false. We need to do this as for the next question we need to reset this value to false. Otherwise this useEffect will not be called from second question onwards as countDownCompleted is in its dependency array and it will always be true.